tests/foreman/api/test_organization.py

```diff
@@ -324,59 +311,30 @@ def test_bugzilla_1103157(self):
         @Assert: Organization contains both compute resources
 
         """
-        compute_resource_ids = [
+        compute_resources = [
             entities.LibvirtComputeResource(
                 name=gen_string('alpha'),
                 url='qemu://host.example.com/system'
-            ).create_json()['id']
+            ).create()
             for _ in range(2)
         ]
-
-        client.put(
-            self.organization.path(),
-            {'organization': {
-                'compute_resource_ids': [compute_resource_ids[0]]
-            }},
-            verify=False,
-            auth=get_server_credentials(),
-        ).raise_for_status()
-
-        self.assertEqual(
-            1,
-            len(self.organization.read_json()['compute_resources']),
-        )
+        self.organization.compute_resource = compute_resources[:1]
+        self.organization = self.organization.update(['compute_resource'])
+        self.assertEqual(len(self.organization.compute_resource), 1)
 
         login = gen_alphanumeric()
         password = gen_alphanumeric()
         user = entities.User(login=login, password=password).create()
-
-        role_id = entities.Role().create_json()['id']
-
+        role = entities.Role().create()
         for perm in ['edit_compute_resources', 'edit_organizations']:
-            permission_ids = [
-                permission['id']
+            permissions = [
+                entities.Permission(id=permission['id'])
                 for permission
                 in entities.Permission(name=perm).search()
             ]
-            entities.Filter(permission=permission_ids, role=role_id).create()
-
-        client.put(
-            user.path(),
-            {u'user': {u'role_ids': [role_id]}},
-            auth=get_server_credentials(),
-            verify=False,
-        ).raise_for_status()
-
-        client.put(
-            self.organization.path(),
-            {'organization': {
-                'compute_resource_ids': [compute_resource_ids[1]]
-            }},
-            verify=False,
-            auth=(login, password),
-        ).raise_for_status()
-
-        self.assertEqual(
-            2,
-            len(self.organization.read_json()['compute_resources']),
-        )
+            entities.Filter(permission=permissions, role=role).create()
+        user.role = [role]
+        user = user.update(['role'])
+        self.organization.compute_resource = compute_resources[1:]
+        self.organization = self.organization.update(['compute_resource'])
```

Comment:
Do you use correct user to update that entity? Main idea here to use not admin user (auth=get_server_credentials()), but just created one (auth=(login, password))